### ISSUE TITLE
Discuss: How to Move an `item` from one `list` to another `list`?

### ISSUE THREAD

**Issue by nelsonic** [dwyl/mvp#420]
Other `list`-based project management tools allow `people` to _easily_ transfer `items` from one `list` to another,
How much do we want to "reinvent the wheel" on this functionality vs. just "copying" what people are already familiar with?

**Comment by LuchoTurtle** [comment]
I don't think we should "reinvent the wheel"/try to create a new UX to change lists. "Copying" what people are already familiar with will provide an easier onboarding experience and reduce initial frustration when first tinkering with the app. 

I see our `lists` in a similar fashion to what happens with Overleaf when tagging documents.

<img width="249" alt="image" src="https://github.com/dwyl/mvp/assets/17494745/905990e0-ae28-4e1d-9f13-ca9b807d5ef1">

We're just using it for **categorization purposes**, but with the added benefit of having specific UX for most used lists (shopping lists, exercise logs, for example).

However, thinking of how to implement this, relates to how we are saving our items in our database.

![image](https://github.com/dwyl/mvp/assets/17494745/f7ee6b9d-aaf0-4ac4-8dd1-5a97f2593912)

As mentioned in https://github.com/dwyl/mvp/issues/418#issuecomment-1712701585, we don't have any relation between the table `items` and `lists`. So changing an item from one list to another means having to change the `seq` field from a list and add it to another. Is there an efficient way of doing this? 💭

**Comment by nelsonic** [comment]
@LuchoTurtle by definition we _always_ want the `items` to be in a `sequence` in the `list`. 
So this is _already_ the most efficient way of doing it. We don't _need_ a relation. `list.seq` works.
If _anything_ is unclear about this approach, please read the code:

https://github.com/dwyl/mvp/blob/b8aee0e3c29d0cb45025595131248c08fbbf6a31/lib/app/list.ex#L149-L154

It couldn't be easier to add an `item` to a `list`. 
I tried several iterations of having a `list_items` table to join the data relationally and they all were more complex than this for maintaining the `ordering` (_sequence_) of the items in the list.

**Comment by nelsonic** [comment]
I should have made it clearer in the OP that this issue is specific to the _interface_ that the `person` interacts with for adding/moving `items` to a `list`.

**Comment by LuchoTurtle** [comment]
Yes, but I was curious about how we were going to do the moving of an item from one list to another (database-wise, not interface). Is there a more efficient way instead other than:

1. open `list.seq`
2. iterate over each `cid`
3. find the `cid` we want to move from
4. splice the `list.seq` array
5. and then add the `cid` to the other `list`?

Regarding the interface itself, as I've said before, no need to reinvent the wheel. For example, look how Trello is doing things (I know it's not a 1-1 comparable application but the flow looks good to me).

<img width="32%" alt="Screenshot 2023-09-13 at 09 22 51" src="https://github.com/dwyl/mvp/assets/17494745/cb666a15-7a0f-46dc-ace0-1ea6b824b154">
<img width="32%" alt="Screenshot 2023-09-13 at 09 22 58" src="https://github.com/dwyl/mvp/assets/17494745/d53314c5-f0e8-4687-9ca6-5e671d4ba195">
<img width="32%" alt="Screenshot 2023-09-13 at 09 22 43" src="https://github.com/dwyl/mvp/assets/17494745/86cdf8e6-0f30-4c1a-ad79-489b6355aced">

They have a modal option (no go) that does the same thing.

<img width="464" alt="Screenshot 2023-09-13 at 09 23 12" src="https://github.com/dwyl/mvp/assets/17494745/b4832696-bcdf-4366-8579-b61152d4cae0">

**Comment by iteles** [comment]
I would agree with you both that this is both a solved problem and such a core piece of most app functionality that 'reinventing the wheel' here might come back to haunt us as feeling 'incorrect' to some people.

The most common way to do this that I know of is essentially through tags (though I would argue I would prefer list and tags to be separate things, but not necessary in early stages). You just add/remove tags and those are the lists the item is part of (from an interface pov).
